File: Daily Test/Python/City Missiles 17.2.2026.py
n = int(input().strip())
cities = []
for _ in range(n):
    x, y = map(int, input().split())
    cities.append((x, y))

positions = {x: y for x, y in cities}
found = False

for x1, y1 in cities:
    target = x1 + y1
    if target in positions:
        y2 = positions[target]
        if target - y2 == x1:
            found = True
            break

if found:
    print("YES")
else:
    print("NO")

File: Daily Test/Java/City Missiles 17.2.2026.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        List<int[]> cities = new ArrayList<>();
        Map<Integer, Integer> positions = new HashMap<>();

        for (int i = 0; i < n; i++) {
            int x = sc.nextInt();
            int y = sc.nextInt();
            cities.add(new int[]{x, y});
            positions.put(x, y);
        }

        boolean found = false;

        for (int[] c : cities) {
            int x1 = c[0], y1 = c[1];
            int target = x1 + y1;

            if (positions.containsKey(target)) {
                int y2 = positions.get(target);
                if (target - y2 == x1) {
                    found = true;
                    break;
                }
            }
        }

        System.out.println(found ? "YES" : "NO");
        sc.close();
    }
}
